runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/FusedPipeline.java

```diff
@@ -102,7 +102,7 @@ static FusedPipeline of(
           generateStageId(
               stage,
               Sets.union(getComponents().getTransformsMap().keySet(), topLevelTransforms.keySet())),
-          stage.toPTransform());
+          stage.toPTransform(getComponents()));
```

Comment:
As I already mentioned, if we're already storing the components within the stage itself, why pass it in here?